src/components/SetUp/FirstForm.jsx

```diff
@@ -0,0 +1,182 @@
+import { useEffect, useState } from 'react';
+import {
+  Box,
+  Button,
+  FormControl,
+  FormLabel,
+  Input,
+  VStack,
+  Heading,
+  Text,
+  Link,
+  HStack,
+  Image,
+  Flex,
+  useToast,
+} from '@chakra-ui/react';
+import { useNavigate } from 'react-router-dom';
+import LOGO from './fph_logo.png';
+import { useAuth } from '../../contexts/AuthContext';
+import { useBackend } from '../../contexts/BackendContext';
+import PropTypes from 'prop-types'; // Import PropTypes
+
+const FirstForm = ({ admin, nextStep }) => {
+  const { isAdmin, currentUser } = useAuth();
+  const { backend } = useBackend();
+  const [email, setEmail] = useState('');
+  const [password, setPassword] = useState('');
+  const [confirmPassword, setConfirmPassword] = useState('');
+  const [id, setId] = useState('');
+  const { signup, logout } = useAuth();
+  const [loading, setLoading] = useState(false);
+  const navigate = useNavigate();
+  const toast = useToast();
+
+  useEffect(() => {
+    const setUp = async () => {
+      const urlParams = new URLSearchParams(window.location.search);
+      const idFromUrl = urlParams.get('id');
+      setId(idFromUrl);
+      await logout();
+    };
+
+    setUp();
+  }, []);
+
+  useEffect(() => {
+    const createUser = async () => {
+      try {
+        await backend.post('/businessUser', { id: id, uid: currentUser.uid });
+        nextStep();
+      } catch (error) {
+        console.error('Error fetching data:', error);
+      }
+    };
+
+    if (currentUser) {
+      createUser();
+    }
+  }, [currentUser]);
```

Comment:
⚠️ **[eslint]** <[react-hooks/exhaustive-deps](https://github.com/facebook/react/issues/14920)> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>React Hook useEffect has missing dependencies: 'backend', 'id', and 'nextStep'. Either include them or remove the dependency array. If 'nextStep' changes too often, find the parent component that defines it and wrap that definition in useCallback.